Support of Modelica_StateGraph2 Connections.uniqueRoot (#3701)
modelica/ModelicaSpecification: chapters/connectors.tex

\end{semantics}
\end{operatordefinition}

\begin{operatordefinition}[Connections.uniqueRoot]
\begin{synopsis}\begin{lstlisting}
Connections.uniqueRoot(a.R, msg)
\end{lstlisting}\end{synopsis}
\begin{semantics}
The overdetermined type or record instance \lstinline!R! in connector instance \lstinline!a! is a \firstuse[definite root node]{(definite) root node}\index{root node!definite} in a virtual connection graph.
The input \lstinline!a! may be an array, in which case the operator is applied to each element of the array.

> Every virtual state diagram graph must have exactly one \lstinline!uniqueRoot! definition (and no normal or potential roots), before breakable branches are removed.

Reviewer: The phrase "virtual state diagram graph" has not been defined.

```suggestion
Every virtual connection graph that contains a \lstinline!uniqueRoot! must have exactly one \lstinline!uniqueRoot! definition (and no normal or potential roots), before breakable branches are removed.
```

It is used for the \lstinline!Modelica_StateGraph2! library.
\end{nonnormative}
\end{semantics}
\end{operatordefinition}

\begin{operatordefinition}[Connections.uniqueRootIndices]
\begin{synopsis}\begin{lstlisting}
Connections.uniqueRoot(a.R, b.R, msg)
\end{lstlisting}\end{synopsis}
\begin{semantics}
The arguments \lstinline!a.R! and \lstinline!b.R! shall be vectors of overdetermined types or record instances, and \lstinline!size(b.R, 1)<=size(a.R, 1)!.
The arguments \lstinline!a.R! must be unique roots.
The result is an integer vector, \lstinline!rootIndices!, of length \lstinline!size(a.R, 1)! containing a permutation of \lstinline!1:size(a.R, 1)! such that:
There is a (unique) path from \lstinline!a[rootIndices[i]].R! to \lstinline!b[i].R! if \lstinline!i<=size(b.R,1)!.
There is no path from \lstinline!a[rootIndices[i]].R! to any \lstinline!b.R! if \lstinline!i>size(b.R,1)!.
In case no such vector exist an error message should be given and include the third argument \lstinline!msg!.

\begin{nonnormative}

> This is used for checking parallel connections in the \lstinline!Modelica_StateGraph2! library.

Author: ```suggestion
This is used for checking that there are delayed transitions in loops involving parallel branches in the \lstinline!Modelica_StateGraph2! library.
```
